### PR TITLE
Onboarding setup page (continues #2822)

### PR DESCRIPTION
**Pull request by paolodamico**
## Changes
- Introduces the setup page from #2822 & https://github.com/PostHog/retention/issues/8.
- Section 1 of the setup page: create your first real project, ingest events and verify installation.

    Screenshot 2021-01-28 at 10 00 27 AM

    

- Changes the project creation logic licensing permissions to exclude demo projects from consideration.
- Fixes a bug in which you’d see the no events ingested warning after verifying your installation.
    
- Adjusts API response of `/api/organization` endpoints to include information on the organization's teams.

**Next PR** will introduce sections 2 & 3 of the setup page.

## Checklist

- [X] All querysets/queries filter by Organization, by Team, and by User
- [X] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

### CODE REVIEW

**Comment by paolodamico** [comment]
Ready for review. For context, there's still the [hot issue](https://posthog.slack.com/archives/C01K4L7U62W/p1611820388000700) to work out.

File: frontend/src/scenes/sceneLogic.ts [hunk 1]
```diff
@@ -106,11 +109,14 @@ export const sceneConfigurations: Partial<Record<Scene, SceneConfig>> = {
...
     [Scene.Personalization]: {
         plain: true,
-        hide_top_nav: true,
+        hideTopNav: true,
     },
     [Scene.Ingestion]: {
         plain: true,
     },
+    [Scene.OnboardingSetup]: {
+        hideDemoWarnings: true,
```
Comment (macobo) [inline frontend/src/scenes/sceneLogic.ts:118]: Should we also do this under project settings?

File: frontend/src/scenes/App.tsx [hunk 2]
```diff
@@ -135,14 +135,13 @@ function _App(): JSX.Element | null {
...
+                    {!sceneConfig.hideTopNav && featureFlags['navigation-1775'] ? <TopNavigation /> : <TopContent />}
                     <Layout.Content className="main-app-content" data-attr="layout-content">
-                        <DemoWarning />
+                        {!sceneConfig.hideDemoWarnings && <DemoWarning />}
 
                         {!featureFlags['hide-billing-toolbar'] && <BillingToolbar />}
                         {featureFlags['navigation-1775'] ? <BackTo /> : null}
-
-                        {currentTeam && !currentTeam.ingested_event && (
+                        {currentTeam && !currentTeam.completed_snippet_onboarding && !sceneConfig.hideDemoWarnings && (
```
Comment (macobo) [inline frontend/src/scenes/App.tsx:144]: I think this conditional is wrong - we only set completed_snippet_onboarding on /ingestion pages. Let's change to `team.ingested_event`?

File: frontend/src/scenes/onboarding/onboardingSetupLogic.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        stepProjectSetup: [
+            () => [userLogic.selectors.demoOnlyProject],
+            (demoOnlyProject: boolean) => {
+                // Step is completed if org has at least one non-demo project
+                return !demoOnlyProject
+            },
+        ],
+        stepInstallation: [
+            () => [organizationLogic.selectors.currentOrganization],
+            (organization: OrganizationType) => organization.any_project_ingested_events,
```
Comment (macobo) [inline frontend/src/scenes/onboarding/onboardingSetupLogic.ts:47]: Kept the previous logic, but I think this is switched with the last step.

It should be any_project_completed_snippet_onboarding -> any_project_ingested_events

File: frontend/src/scenes/teamLogic.tsx [hunk 4]
```diff
@@ -34,17 +33,11 @@ export const teamLogic = kea<teamLogicType<TeamType>>({
...
-            let location = '/ingestion'
-            if (userLogic.values.user?.organization?.teams) {
-                for (const team of userLogic.values.user.organization.teams) {
-                    if (!team.is_demo && team.id !== values.currentTeam?.id) {
-                        /* If organization already has another non-demo project setup, take to settings, otherwise take to
-                        ingestion wizard */
-                        location = '/project/settings'
-                    }
-                }
+            if (values.currentTeam) {
```
Comment (paolodamico) [inline frontend/src/scenes/teamLogic.tsx:36]: @macobo not sure if this makes sense anymore, a project will always have `completed_snippet_onboarding = False` after creation (per your latest change). Seems like the logic here could be as straightforward as redirecting to `/ingestion`?

Comment (macobo) [reply frontend/src/scenes/teamLogic.tsx:36]: You're right

Comment (paolodamico) [reply frontend/src/scenes/teamLogic.tsx:36]: Cool thanks will change myself, need to do one final tweak before merging

File: frontend/src/scenes/onboarding/onboardingSetupLogic.ts (hunk 3 above)
Comment (paolodamico) [reply frontend/src/scenes/onboarding/onboardingSetupLogic.ts:47]: I think the original logic is right. `ingested_event` is set as soon as an event is received (albeit async) while `completed_snippet_onboarding` is set on the verification step [after the user clicks continue] (see https://github.com/PostHog/posthog/blob/15d04d9e568ed51cb9e217f93c48b0182b536cbf/frontend/src/scenes/ingestion/ingestionLogic.ts#L103)

File: frontend/src/scenes/App.tsx (hunk 2 above)
Comment (paolodamico) [reply frontend/src/scenes/App.tsx:144]: You're right! I think we need to have both cases. Addressing in subsequent PR.

File: yarn.lock [hunk 5]
```diff
@@ -4015,9 +4015,9 @@ date-fns@^2.0.1, date-fns@^2.15.0:
   integrity sha512-sAJVKx/FqrLYHAQeN7VpJrPhagZc9R4ImZIWYRFZaaohR3KzmuK88touwsSwSVT8Qcbd4zoDsnGfX4GFB4imyQ==
```
[the remark this answers is not in the document]
Comment (paolodamico) [reply yarn.lock:4015]: Ran yarn upgrade on Ant, updated to latest packages of some dependencies.

**Pull request merged**